test/unit/serializer.cpp

```diff
@@ -256,30 +263,31 @@ struct serializer_test
         check_src(
             "HTTP/1.1 200 OK\r\n"
             "Server: test\r\n"
-            "Content-Length: 5\r\n"
+            "Content-Length: 2048\r\n"
             "\r\n",
-            test_source{},
-            //--------------------------
+            test_source{std::string(2048, 'A')},
             "HTTP/1.1 200 OK\r\n"
             "Server: test\r\n"
-            "Content-Length: 5\r\n"
-            "\r\n"
-            "12345");
+            "Content-Length: 2048\r\n"
+            "\r\n" +
+            std::string(2048, 'A'));
 
         // source chunked
         check_src(
             "HTTP/1.1 200 OK\r\n"
             "Server: test\r\n"
             "Transfer-Encoding: chunked\r\n"
             "\r\n",
-            test_source{},
-            //--------------------------
+            test_source{std::string(1024, 'A')},
             "HTTP/1.1 200 OK\r\n"
             "Server: test\r\n"
             "Transfer-Encoding: chunked\r\n"
             "\r\n"
-            "0000000000000005\r\n"
-            "12345"
+            "0000000000000367\r\n" +
+            std::string(0x367, 'A') +
+            "\r\n"
+            "0000000000000099\r\n" +
+            std::string(0x099, 'A') +
```

Comment:
The chunk sizes depend on the serializer implementation (i.e., the amount of buffer available when calling source::read). One way to avoid using explicit numbers would be to create a basic chunk decoder and assert only the expected number of chunks and their content.
I'm looking for ideas to see if there is a simpler way of doing it.